streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableAggregateTest.java

```diff
@@ -125,23 +118,43 @@ public void testAggCoalesced() {
         final KTable<String, String> table2 = table1
             .groupBy(
                 MockMapper.noOpKeyValueMapper(),
-                stringSerialzied)
+                stringSerialized)
             .aggregate(MockInitializer.STRING_INIT,
                 MockAggregator.TOSTRING_ADDER,
                 MockAggregator.TOSTRING_REMOVER,
                 Materialized.<String, String, KeyValueStore<Bytes, byte[]>>as("topic1-Canonized")
                     .withValueSerde(stringSerde));
 
+        final MockProcessorSupplier<String, Object> supplier = new MockProcessorSupplier<>();
         table2.toStream().process(supplier);
 
-        driver.setUp(builder, stateDir);
-
-        driver.process(topic1, "A", "1");
-        driver.process(topic1, "A", "3");
-        driver.process(topic1, "A", "4");
-        driver.flushState();
-
-        assertEquals(Collections.singletonList("A:0+4 (ts: 0)"), supplier.theCapturedProcessor().processed);
+        try (
+            final TopologyTestDriver driver = new TopologyTestDriver(
+                builder.build(),
+                mkProperties(mkMap(
+                    mkEntry(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy"),
+                    mkEntry(StreamsConfig.APPLICATION_ID_CONFIG, "test"),
+                    mkEntry(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory("kafka-test").getAbsolutePath())
+                )),
+                0L)) {
+            final ConsumerRecordFactory<String, String> recordFactory =
+                new ConsumerRecordFactory<>(new StringSerializer(), new StringSerializer(), 0L, 0L);
+
+            driver.pipeInput(recordFactory.create(topic1, "A", "1"));
+            driver.pipeInput(recordFactory.create(topic1, "A", "3"));
+            driver.pipeInput(recordFactory.create(topic1, "A", "4"));
+
+            assertEquals(
+                asList(
+                    "A:0+1 (ts: 0)",
+                    "A:0+1-1 (ts: 0)",
+                    "A:0+1-1+3 (ts: 0)",
+                    "A:0+1-1+3-3 (ts: 0)",
+                    "A:0+1-1+3-3+4 (ts: 0)"
```

Comment:
This difference is a mystery to me. We previously expected `A:0+4 (ts: 0)`. In other words, the processor never even saw the intermediate records. I can't figure out the mechanism for this, so there's a risk we're not testing the right thing here anymore. Help?

Reply:
I'm not too (at all) familiar with either driver, but isn't the difference just that, like above, we're now flushing on every record and see all the intermediates, whereas before we were only flushing after processing all three records here?

Reply:
Yeah, but what seems strange to me is that the processor should still have seen all the events, and then caching should have absorbed some of the intermediate ones, but looking at the old output, it seems that the processor never saw those intermediate events, which indicates that there was some caching upstream of the processor. There's only a source KTable, but it's not materialized. I was thinking that it should not actually be stored in that case, but now that I'm reflecting on it again, I think that's only if you have optimizations enabled. So, this would explain it.

Reply:
The source KTable is materialized for this case. That's why `groupBy().aggregate()` is only executed after the flush on the last value per key.

However, with `TopologyTestDriver` we cannot mimic this behavior any longer. Hence, it would make sense to delete this test, as it's the same as `testAggBasic`

Reply:
Roger, wilco.